src/zmac.y

```diff
@@ -2318,50 +2338,34 @@ statement:
 	}
 |
 	IF_TK expression '\n' {
+		do_if($2);
+	}
+|
+	IFE_TK expression '\n' {
 		expr_number_check($2);
-		if (ifptr >= ifstmax)
-			error("Too many ifs");
-		else
-			*++ifptr = !($2->e_value);
-
-		saveopt = fopt;
-		fopt = 1;
-		list($2->e_value);
-		fopt = saveopt;
+		do_if_value(!$2->e_value);
 		expr_free($2);
 	}
+|
+	IF1_TK '\n' {
+		do_if_value(npass == 1);
+	}
+|
+	IF2_TK '\n' {
+		do_if_value(npass == 2);
```

Comment:
The original mac assembler was two pass so "IF2" was true on the output pass.  So I think it would be better if this were `do_if_value(outpass)`  Should be equivalent for original source code and more useful in new code.

If that change breaks your tests then we can leave it and worry about it later.

Reply:
I went back and forth on this but didn't reach a thorough conclusion. `IF2` seems to be often used around print statements. If I use `outpass`, then those statements will be printed N-1 times, N being the total number of passes.

But you're right, if the user is wrapping something besides prints, it won't work correctly. Since I doubt many folks are actually using this to generate printouts these days, I think we should go with your suggestion and use `outpass` instead. Agreed?

Reply:
I agree, go with `outpass`.  A comment about the choice would be good.

As an aside, I'm surprised that `outpass` causes multiple printings.  I thought it was only set when zmac had decided the assembly had stabilized and it is time to write the output.